data/src/main/java/com/linkedin/data/schema/resolver/JarFileDataSchemaLocation.java

```diff
@@ -0,0 +1,69 @@
+/*
+   Copyright (c) 2012 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.data.schema.resolver;
+
+import com.linkedin.data.schema.DataSchemaLocation;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.ZipEntry;
+
+
+/**
+ * Represents the location of a DataSchema file in a Jar.
+ * Similar to {@link InJarFileDataSchemaLocation}, but it does not contain the jar file itself
+ */
+public class JarFileDataSchemaLocation implements DataSchemaLocation {
+  private final String _jarFileName;
+  private final String _pathInJar;
+
+  public JarFileDataSchemaLocation(String jarFileName, String pathInJar)
+  {
+    _jarFileName = jarFileName;
+    _pathInJar = pathInJar;
+  }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o)
+      return true;
+    if (o instanceof JarFileDataSchemaLocation == false)
```

Comment:
This would break the functionality when a location is checked agains known bad locations.

In that case InJarFileDataSchemaLocation will be checked and it would not match JarFileDataSchemaLocation

Reply:
Updated to account for this and added unit tests